### ISSUE TITLE
#[inline] should be a warning or error if used above non-functions

### ISSUE THREAD

**Issue by mtak-** [rust-lang/rust#45568]
Adding `#[inline]` above arbitrary statements does not issue a warning or error.

I found this confusing. I was guessing at a loop unrolling attribute when I discovered this, and thought the lack of warnings/errors meant it might actually unroll the loop, but checking the ASM output proves it does not: https://godbolt.org/g/sNEnfa

```rust
#[inline]
fn foo() {} // ok

fn bar() {
  #[inline] // pointless, but no warning/error
  ()
}
```

**Comment by zackmdavis** [comment]
Could be part of the unused-attribute lint? Potentially related: #14407 and #14408

**Comment by oyvindln** [comment]
Possibly a duplicate of #43988 ?

**Comment by mtak-** [comment]
@oyvindln Oops, I'd say it is definitely a dupe your issue (#43988). closing
